practice-one/src/scripts/views/book-detail.view.js

```diff
@@ -2,109 +2,114 @@ import BookModel from "../models/book.model";
+            }
+        });
+    }
+
+    displayBookInfo(bookInfo) {
+        const bookTitleElement = getElementById("book-title");
+        const bookAuthorElement = getElementById("book-author");
+        const bookDateElement = getElementById("book-date");
+        const bookDescriptionElement = getElementById("book-description");
+        const bookCreateElement = getElementById("book-create");
+        const bookUpdateElement = getElementById("book-update");
+        const bookImage = getElementById("img-detail-wp-render");
+
+        bookTitleElement.textContent = bookInfo.bookname;
+        bookAuthorElement.textContent = `Author: ${bookInfo.author}`;
+        bookDateElement.textContent = `Published Date: ${bookInfo.date}`;
+        bookDescriptionElement.textContent = bookInfo.description;
+        bookCreateElement.textContent = `Create At: ${bookInfo.date}`;
+        bookUpdateElement.textContent = `Update At: ${bookInfo.date}`;
+        bookImage.innerHTML = `<img class="image-detail" src="${bookInfo.image}"/>`;
+    }
+
+    setupEditButton(bookInfo) {
+        const editButton = querySelector(".edit-detail");
+        const validationForm = getElementById("validation-form");
+        const overlay = getElementById("overlay");
+
+        editButton.addEventListener("click", () => {
+            getElementById("bookname").value = bookInfo.bookname;
+            getElementById("author").value = bookInfo.author;
+            getElementById("date").value = bookInfo.date;
+            getElementById("description").value = bookInfo.description;
+            getElementById("preview-link-image").innerText = bookInfo.image;
+            validationForm.style.display = "block";
+            overlay.style.display = "block";
+        });
+    }
+
+    setupCancelButton() {
+        const cancelButton = querySelector(".cancel");
+        const validationForm = getElementById("validation-form");
+        const overlay = getElementById("overlay");
+
+        cancelButton.addEventListener("click", () => {
+            validationForm.style.display = "none";
+            overlay.style.display = "none";
+        });
+    }
+
+    setupSaveButton(bookInfo) {
+        const saveButton = querySelector(".save");
+        const validationForm = getElementById("validation-form");
+        const overlay = getElementById("overlay");
+
+        saveButton.addEventListener("click", () => {
+            const updatedBookInfo = {
+                bookname: getElementById("bookname").value,
+                author: getElementById("author").value,
+                date: getElementById("date").value,
+                description: getElementById("description").value,
+            };
+
+            const bookModel = new BookModel();
+            bookModel.updateBookByInfo(bookInfo.id, updatedBookInfo);
+
+            this.displayBookInfo(updatedBookInfo);
+
+            validationForm.style.display = "none";
```

Comment:
Let think about how to refactor this code, can reuseable
I saw `style.display = "none"` repeat a lot